Data-Science/how-to-find-problem-views-from-profiling.md

```diff
@@ -0,0 +1,121 @@
+# How to find Problem Views from Profiling
+
+## Question
+
+I have a very complex data model with many layers of views.
+In query profiling I identified a problematic join between two tables.
+
+How do I find out which view(s) lead to the inclusion of those tables, so I can try and fix things?
+
+
+## Answer
+
+### Example Setup
+
+As is often the case with dependency tracking, we require a top-level view to start our search.
+In the following **example**, we will use the view `CUSTOMER_STATS_RLS.F_MONITOR` as entry point.
+
+Selecting from the view will provide a profile which includes the following pipeline:
+
+|PART_ID|PART_NAME|PART_INFO|OBJECT_SCHEMA|OBJECT_NAME|
+|---|---|---|---|---|
+|7|PIPE SCAN|on REPLICATED table|CDP|DATABASE_JI|
+|8|PIPE JOIN|(null)|CDP|ISSUES|
+|9|PIPE JOIN|on REPLICATED table|CDP|ISSUESTATI|
+|10|PIPE FILTER|(null)|(null)|(null)|
+|11|PIPE INDEX OUTER JOIN|on REPLICATED table|CDP|ISSUEVERSIONS|
+|12|PIPE PREFERENCE PREFILTER|(null)|(null)|(null)|
+|13|PIPE FULL PREFETCH|PIPE BUFFER|CDP|ISSUES|
+
+Unfortunately for us, profiling shows the final execution plan, which contains tables only, the view layer has
+been stripped away.
+
+Let's say we are interested how `ISSUES` is used twice, and how it relates to `DATABASE_JI`.
+
+### Analysis Query
+
+We will use the following query (template) to perform dependency analysis:
+```sql
+with k1 as (
+    select
+        sys_connect_by_path(object_schema||'.'||object_name, ' > ') as pth,
+        referenced_object_schema,
+        referenced_object_name
+    from sys.exa_all_dependencies
```

Comment:
Maybe [EXA_DBA_DEPENDENCIES​_RECURSIVE](https://docs.exasol.com/db/latest/sql_references/system_tables/metadata/exa_dba_dependencies_recursive.htm) could be useful here, to hide CONNECT BY complexity.

Reply:
Unfortunately, the recursive table is useless if you want to find the full path leading from source to target.